lib/BackbeatConsumer.js

```diff
@@ -381,8 +382,8 @@ class BackbeatConsumer extends EventEmitter {
         // needs explicit polling to get new messages
 
         const nNewConsumeRequests = this._getAvailableSlotsInPipeline();
-        if (nNewConsumeRequests === 0) {
-            // processing pipeline is 100% busy, do not attempt to consume
+        if (nNewConsumeRequests === 0 || this._isShuttingDown) {
```

Comment:
should we use a flag (as you did), or `pause()` the consumer?
the risk would be that we fail to poll, and get kicked out of the group without waiting for us to commit...

Reply:
i don't think having the consumer getting kicked is an issue here, the only case where the `close()` function gets called is when the pods shutdown.

Using the pause function already implemented is probably a bad idea as it unsubscribes from the topic, so the consumer won't be able to commit.

Using the Kafka consumer's pause function is going to add more complexity that is not really needed, as we have to track the paused topic/partitions and update them in case of an assignment from a rebalance, ...

Reply:
I was talking about consumer.pause()

Reply:
> as we have to track the paused topic/partitions and update them in case of an assignment from a rebalance

Do we? This is shutdown, so we should not take part in the assign...

Btw, maybe we should first unsubscribe: then wait for revoke to complete (eg drain, commi, unassign) : then disconnect and quit...

Reply:
good idea, we should use the logic already present in the rebalance callback to wait for jobs and commit the offsets